pkgs/applications/misc/dstask/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "dstask";
+  version = "0.18";
+  rev = "d687a0f9aec634a808cfa0df4aecb06ab4f57732";
+
+  src = fetchFromGitHub {
+    owner = "naggie";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "16z5zlfj955pzsj0l58835slvpchdaq2vbyx2fjzi6y9xn1z2nd1";
+  };
+
+  vendorSha256 = null;
+
+  buildFlagsArray = [ ''-ldflags=-w -s
+    -X "github.com/naggie/dstask.VERSION=v${version}"
+    -X "github.com/naggie/dstask.GIT_COMMIT=${rev}"
```

Comment:
What are these flags doing? Why are they needed? What does `rev` specify and why is it different from the source `rev`?

Reply:
@foxit64 Would you be able to assist me here?

Reply:
The ldflags reduce the executable size by stripping some debug stuff.

The variables are set so that the output of `dstask version` shows the git ref and the release version from github.

Reply:
thanks naggie for the explanation.

Reply:
Thanks naggie!:)

Reply:
So basically `ref` is just the fully-qualified commit hash of the tag `version`? It seems likely that someone will forget to update that in the future, especially since it will not lead to runtime breakage. If it is really necessary to encode that information, the `ref` definition should be close to `version` (as it already is) and there should be a comment that explains how `ref` should be updated when updating the package.

I generally don't see much point in duplicating this information though, since the version should convey the same information right? If we manually have to maintain that, it makes updates a little bit more cumbersome. More importantly, all updates will need to be done manually since our automatic update infrastructure (https://github.com/ryantm/nixpkgs-update) won't know about this.

Reply:
It would be very nice if the package could be automatically updated with [nixpkgs-update](https://github.com/ryantm/nixpkgs-update)! What do you think, @naggie? Would [this code](https://github.com/naggie/dstask/blob/2921be9d9c300d963b79f0d4ceb7795c396aa8d4/cmd/dstask.go#L333) crash if we didn't supply the `GIT_COMMIT` build flag?